src/test/java/io/specto/hoverfly/junit/core/HoverflyTest.java

```diff
@@ -32,8 +35,12 @@
     private Hoverfly hoverfly;
     private ObjectMapper mapper = new ObjectMapper();
 
+    @Rule
+    public final RestoreSystemProperties restoreSystemProperties = new RestoreSystemProperties();
```

Comment:
You're right, Hoverfly should restore system properties in its own cleanup method. I will raise an issue for this bug.